embedded/tbtree/tbtree.go

```diff
@@ -1073,48 +1089,82 @@ func (t *TBtree) flushTree(synced bool) (wN int64, wH int64, err error) {
 
 	cLogEntry.nLogChecksum, err = appendable.Checksum(t.nLog, t.committedNLogSize, wN)
 	if err != nil {
-		return 0, 0, t.wrapNwarn("Flushing index '%s' {ts=%d} returned: %v", t.path, t.root.ts(), err)
+		return 0, 0, t.wrapNwarn("Flushing index '%s' {ts=%d, cleanup_percentage=%d} returned: %v",
+			t.path, t.root.ts(), cleanupPercentage, err)
 	}
 
 	cLogEntry.hLogChecksum, err = appendable.Checksum(t.hLog, t.committedHLogSize, wH)
 	if err != nil {
-		return 0, 0, t.wrapNwarn("Flushing index '%s' {ts=%d} returned: %v", t.path, t.root.ts(), err)
+		return 0, 0, t.wrapNwarn("Flushing index '%s' {ts=%d, cleanup_percentage=%d} returned: %v",
+			t.path, t.root.ts(), cleanupPercentage, err)
 	}
 
 	_, _, err = t.cLog.Append(cLogEntry.serialize())
 	if err != nil {
-		return 0, 0, t.wrapNwarn("Flushing index '%s' {ts=%d} returned: %v", t.path, t.root.ts(), err)
+		return 0, 0, t.wrapNwarn("Flushing index '%s' {ts=%d, cleanup_percentage=%d} returned: %v",
+			t.path, t.root.ts(), cleanupPercentage, err)
 	}
 
 	err = t.cLog.Flush()
 	if err != nil {
-		return 0, 0, t.wrapNwarn("Flushing index '%s' {ts=%d} returned: %v", t.path, t.root.ts(), err)
+		return 0, 0, t.wrapNwarn("Flushing index '%s' {ts=%d, cleanup_percentage=%d} returned: %v",
+			t.path, t.root.ts(), cleanupPercentage, err)
 	}
 
 	t.insertionCountSinceFlush = 0
-	t.log.Infof("Index '%s' {ts=%d} successfully flushed", t.path, t.root.ts())
+	t.log.Infof("Index '%s' {ts=%d, cleanup_percentage=%d} successfully flushed",
+		t.path, t.root.ts(), cleanupPercentage)
 
 	if sync {
 		err = t.cLog.Sync()
```

Comment:
nLog and hLog should be synced too

Reply:
yes, there are synced before writing into the commit log. Sync could be right before cLog sync as well...

Reply:
OK, didn't see previous calls to Sync - those are in different part of the code.